src/app/domain/geolocation/tracking/tracking.js

```diff
@@ -402,25 +404,43 @@ export const filterNonHeadingPointsAfterStillActivity = (
   return points
 }
 
-const getActivitiesFromLocations = locations => {
-  if (!locations || locations.length < 1) {
-    return null
-  }
-  const activities = locations.map(loc =>
-    translateEventToEMissionMotionActivity(loc)
+/**
+ * We noticed that some "stationary" activities were in fact in motion
+ * This detects this strange behaviour that we experienced on some Android devices
+ *
+ * @param {Location} location - The location point
+ * @returns {boolean} whether or not this a moving stationary activity
+ */
+const isMovingStill = location => {
+  return (
+    (location.activity.type === STILL_ACTIVITY && location.is_moving) ||
+    (location.activity.type === STILL_ACTIVITY &&
+      location?.coords?.speed > 0 &&
+      location?.coords?.speed_accuracy > 0)
   )
-  return activities
 }
 
 export const getFilteredActivities = async ({ beforeTs, locations }) => {
-  let activities = await getActivities({ beforeTs })
-  if (!activities || activities?.length < 1) {
-    Log('No activity found in local storage. Get it from locations')
-    // Fallback when no activity was captured
-    // TODO: filter stationary
-    const activitiesFromLoc = getActivitiesFromLocations(locations)
-    return activitiesFromLoc
+  const savedActivities = await getActivities({ beforeTs })
+  let activitiesFromLocations = []
+  if (locations && locations.length > 0) {
+    activitiesFromLocations = locations.map(loc => {
+      const location = { ...loc }
```

Comment:
What is the purpose of cloning the object here ?

Reply:
To avoid mutating the original object, as doing so can be sometimes tedious